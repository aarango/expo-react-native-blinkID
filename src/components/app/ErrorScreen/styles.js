import { StyleSheet } from 'react-native';
import { Colors } from '@auteco/theme';

const styles = StyleSheet.create({
  button: {
    alignSelf: 'center',
    backgroundColor: Colors.PRIMARY,
    borderRadius: 5,
    height: 50,
    justifyContent: 'center',
    marginTop: 28,
    width: 200,
  },
  buttonText: {
    color: Colors.WHITE,
    fontSize: 18,
    fontWeight: 'bold',
  },
  container: {
    flex: 1,
  },
  errorText: {
    alignSelf: 'center',
    color: Colors.GRAY_LIGHT,
    fontSize: 14,
    marginTop: 15,
  },
  header: {
    alignItems: 'center',
    backgroundColor: Colors.BLACK,
    flexDirection: 'row',
    height: 60,
    width: '100%',
  },
  iconContainer: {
    alignItems: 'center',
    height: 50,
    justifyContent: 'center',
    marginHorizontal: 20,
    marginTop: 3,
    width: 50,
  },
  image: {
    height: 50,
    width: 50,
  },
  mainText: {
    alignSelf: 'center',
    fontSize: 18,
    width: '80%',
  },
  safeArea: {
    flex: 1,
  },
  textCentred: {
    textAlign: 'center',
  },
  title: {
    color: Colors.BLACK,
    fontSize: 25,
    fontWeight: 'bold',
    lineHeight: 24,
    marginBottom: 24,
  },
  verticallyCentered: {
    justifyContent: 'center',
  },
});

export { styles };
