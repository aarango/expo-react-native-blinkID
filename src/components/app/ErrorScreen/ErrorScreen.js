import React from 'react';
import { Colors } from '@auteco/theme';
import { styles } from './styles';
import { header as images } from '@auteco/assets';
import { View, Text, Image, SafeAreaView, StatusBar } from 'react-native';
import PropTypes from 'prop-types';

const ErrorScreen = ({ error }) => {
  return (
    <>
      <SafeAreaView style={styles.safeArea}>
        <StatusBar barStyle="dark-content" backgroundColor={Colors.WHITE} />
        <View style={styles.header}>
          <View style={styles.iconContainer}>
            <Image
              style={styles.image}
              source={images.logo}
              resizeMode="contain"
            ></Image>
          </View>
        </View>
        <View
          style={[styles.container, styles.verticallyCentered]}
          testID="NoConnection"
        >
          <Text style={[styles.textCentred, styles.title]}>Discúlpanos</Text>
          <Text style={[styles.textCentred, styles.mainText]}>
            Se ha presentando un error inesperado.
          </Text>
          <Text style={[styles.textCentred, styles.mainText]}>
            Por favor cierra la aplicación he intenta de nuevo.
          </Text>
          {error && (
            <Text
              testID="error-text"
              style={[styles.textCentred, styles.errorText]}
            >
              Error: {error.message}
            </Text>
          )}
        </View>
      </SafeAreaView>
    </>
  );
};

ErrorScreen.propTypes = {
  error: PropTypes.object,
};

export { ErrorScreen };
