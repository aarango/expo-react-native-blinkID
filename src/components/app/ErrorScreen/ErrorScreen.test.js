import React from 'react';
import { render } from '@testing-library/react-native';
import { ErrorScreen } from './ErrorScreen';

describe('<ErrorScreen />', () => {
  it('should match the snapshot', async () => {
    const error = { message: 'The error' };
    const { toJSON, getByTestId } = render(<ErrorScreen error={error} />);
    expect(getByTestId('error-text')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
  it('should match the snapshot', async () => {
    const error = undefined;
    const { queryByTestId } = render(<ErrorScreen error={error} />);
    expect(queryByTestId('error-text')).toBeFalsy();
  });
});
