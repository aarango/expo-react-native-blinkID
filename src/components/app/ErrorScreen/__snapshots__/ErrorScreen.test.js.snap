// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorScreen /> should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexDirection": "row",
        "height": 60,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginTop": 3,
          "width": 50,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
    testID="NoConnection"
  >
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#000000",
            "fontSize": 25,
            "fontWeight": "bold",
            "lineHeight": 24,
            "marginBottom": 24,
          },
        ]
      }
    >
      Discúlpanos
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "width": "80%",
          },
        ]
      }
    >
      Se ha presentando un error inesperado.
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "width": "80%",
          },
        ]
      }
    >
      Por favor cierra la aplicación he intenta de nuevo.
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "alignSelf": "center",
            "color": "#A4A6AB",
            "fontSize": 14,
            "marginTop": 15,
          },
        ]
      }
      testID="error-text"
    >
      Error: 
      The error
    </Text>
  </View>
</RCTSafeAreaView>
`;
